llvm/lib/Transforms/Vectorize/VPlanTransforms.cpp

```diff
@@ -2250,6 +2280,13 @@ static void licm(VPlan &Plan) {
   }
 }
 
+/// Move loop-invariant recipes out of the vector loop region in \p Plan.
+static void licm(VPlan &Plan) {
+  VPRegionBlock *LoopRegion = Plan.getVectorLoopRegion();
+  sinkRecipesFromLoopRegion(LoopRegion);
+  hoistRecipesFromLoopRegion(LoopRegion);
```

Comment:
```suggestion
  licmSink(Plan);
  licmHoist(Plan);
```

Not sure if this function is necessary: maybe inline in optimize?